Engine/ResourceManagement/Manager/TextureManager.cpp

```diff
@@ -15,45 +18,60 @@
 #include <IL/ilu.h>
 #include <IL/ilut.h>
 
+constexpr size_t extension_size = 3; //3 characters: DDS, TGA, JPG..
 std::shared_ptr<Texture> TextureManager::Load(uint32_t uuid, const FileData& resource_data)
 {
-	std::shared_ptr<Texture> loaded_texture;
+	std::string extension;
+	TextureOptions texture_options;
+	char* cursor = (char*)resource_data.buffer;
+	size_t bytes = sizeof(TextureOptions);
+	memcpy(&texture_options, cursor, bytes);
+
+	cursor += bytes;
+	bytes = extension_size;
+	extension.resize(extension_size);
+	memcpy( extension.data(), cursor, bytes);
+
+	size_t offset = extension_size + sizeof(TextureOptions);
 
-	if (MetafileManager::GetUUIDExportedFile(uuid).find("_normal") != std::string::npos)
+	std::vector<char> data;
+	int width, height;
+	bool normal_map = texture_options.texture_type == TextureType::NORMAL;
```

Comment:
here you assing that the texture is normal_map, but where you check if the new given texture is a normal map? I can't find it sorry

Reply:
I will be passing to the texture the hold texture_options. https://github.com/OnionGalaxy/LittleOrionEngine/pull/86/files/341434363918c9689596efb90324717cd37295e5#diff-7a82ea531a622a047e8decf845ca23e6R54 so you can add new texure options and use them when loading the texture